htdocs/core/triggers/interface_50_modTicket_TicketEmail.class.php

```diff
@@ -399,9 +399,11 @@ private function composeAndSendCustomerMessage($sendto, $base_subject, $body, $s
 			$result = $mailfile->sendfile();
 			if ($result) {
 				// update last_msg_sent date
-				$object->fetch($object->id);
+				$save_extrafields = $object->array_options;
```

Comment:
The fetch just after contains the load of the ->array_options so i don't see the need to save it.

Reply:
The problem source is in Ticket::create() method. In this method, triggers are called (l.493) for ticket creation *before* extrafields values are registered (l.510).

Here, on a ticket creation :
- at line 402, `$object->array_options` is set , but extrafields values are not registered in the DB yet.
- at line 403, calling `$object->fetch($object->id)` replaces  `$object->array_options` by an empty array.
- When returning in Ticket::create() method, `$object->array_options` is empty and extrafields values can't be saved.
- As a consequence, saving and restoring `array_options` property is necessary on ticket creation.

We need  `$object->fetch($object->id)` at line 403 when a ticket is being closed, otherwise the `$object->update()` at line 405 will restore its state to the former state.

Reply:
Another solution would be in Ticket::create() to move the ->insertExtrafields() call before the ->call_trigger(). What do you prefer ?

Reply:
move the ->insertExtrafields() call before the ->call_trigger() !!!
triggers on create must always be called at end. The createCommon() used for modules build with modulebuilder rules (see this method for example on good sequence).

Reply:
Thanks !